src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -494,35 +494,263 @@ private static bool HexNumberToBigInteger(ref BigNumberBuffer number, out BigInt
+                        Span<uint> newBuffer = new uint[bufferSize];
+
+                        arrayFromPool = ArrayPool<int>.Shared.Rent(1);
+                        Span<uint> multiplier = MemoryMarshal.Cast<int, uint>(arrayFromPool);
+                        multiplier[0] = TenPowMaxPartial;
+
+                        int blockSize = 1;
+                        while (true)
+                        {
+                            fixed (uint* bufPtr = buffer, newBufPtr = newBuffer, mulPtr = multiplier)
+                            {
+                                uint* curBufPtr = bufPtr;
+                                uint* curNewBufPtr = newBufPtr;
+                                // merge each block pairs.
+                                // When buffer represents:
+                                // |     A     |     B     |     C     |     D     |
+                                // Make newBuffer like:
+                                // |  A + B * multiplier   |  C * multiplier + D   |
+                                for (int i = 0; i < bufferSize; i += blockSize * 2)
+                                {
+                                    int len = Math.Min(bufferSize - i, blockSize * 2);
+                                    int lowerLen = Math.Min(len, blockSize);
+                                    int upperLen = len - lowerLen;
+                                    if (upperLen != 0)
+                                    {
+                                        BigIntegerCalculator.Multiply(mulPtr, blockSize, curBufPtr + blockSize, upperLen, curNewBufPtr, len);
+                                    }
+
+                                    long carry = 0;
+                                    int j = 0;
+                                    for (; j < lowerLen; j++)
+                                    {
+                                        long digit = (curBufPtr[j] + carry) + curNewBufPtr[j];
+                                        curNewBufPtr[j] = unchecked((uint)digit);
+                                        carry = digit >> 32;
+                                    }
+                                    if (carry != 0)
+                                    {
+                                        while (true)
+                                        {
+                                            curNewBufPtr[j]++;
+                                            if (curNewBufPtr[j] != 0)
+                                            {
+                                                break;
+                                            }
+                                            j++;
+                                        }
+                                    }
+
+                                    curBufPtr += blockSize * 2;
+                                    curNewBufPtr += blockSize * 2;
+                                }
+                            }
+
+                            var tmp = buffer;
```

Comment:
Style: Bad use of var.

```suggestion
                            Span<uint> tmp = buffer;
```